File: Arrays/Easy/1512/1512.py
class Solution:
    def numIdenticalPairs(self, nums: List[int]) -> int:
        output: int = 0
        hash: dict[int, int] = {}
        
        for num in nums:
            if num not in hash:
                hash[num] = 1
            else:
                hash[num] += 1
        
        for count in hash.values():
            if count > 1:
                output += (count * (count - 1)) // 2
        
        return output

File: Arrays/Easy/1512/1512.ts
function numIdenticalPairs(nums: number[]): number {
  let output = 0;
  const countMap: Record<number, number> = {};

  for (const num of nums) {
    if (countMap[num] === undefined) {
      countMap[num] = 1;
    } else {
      countMap[num]++;
    }
  }

  for (const count of Object.values(countMap)) {
    if (count > 1) {
      output += (count * (count - 1)) / 2;
    }
  }

  return output;
}
